IdentityCore/src/cache/accessor/MSIDDefaultTokenCacheAccessor.m

```diff
@@ -1035,8 +1069,28 @@ - (BOOL)saveAccount:(MSIDAccount *)account
     
     for (MSIDAccountCacheItem *accountCacheItem in allAccounts)
     {
-        // If we have accountIds to filter by, only return account if it has an associated refresh token
-        if ([accountIds containsObject:accountCacheItem.homeAccountId])
+        BOOL shouldReturnAccount = NO;
+        
+        if (!signedInAccountsOnly)
+        {
+            shouldReturnAccount = YES;
+        }
+        else if ([accountIds containsObject:accountCacheItem.homeAccountId])
+        {
+            shouldReturnAccount = YES;
+        }
+        
+        MSIDAccountMetadataState signInState = [accountMetadataCache signInStateForHomeAccountId:accountCacheItem.homeAccountId clientId:clientId context:nil error:nil];
```

Comment:
nit optimization: This query is only needed if (signedInAccountsOnly == YES), right?

Reply:
yes, addressed ;)